code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -408,3 +498,243 @@
+	desc = "An industry-grade Impressa Modello 5 Coffeemaker of the Piccionaia Home Appliances premium coffeemakers product line. Makes coffee from fresh dried whole beans."
+	icon = 'icons/obj/machines/coffeemaker.dmi'
+	icon_state = "coffeemaker_impressa"
+	circuit = /obj/item/circuitboard/machine/coffeemaker/impressa
+	initial_cartridge = null		//no cartridge, just coffee beans
+	brew_time = 15 SECONDS			//industrial grade, its faster than the regular one
+	density = TRUE
+	pass_flags = PASSTABLE
+	#define BEAN_CAPACITY 10	//amount of coffee beans that can fit inside
+	/// Current amount of coffee beans stored
+	var/coffee_amount = 0
+	/// List of coffee bean objects are stored
+	var/list/coffee = list()
+
+/obj/machinery/coffeemaker/impressa/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		coffeepot = new /obj/item/reagent_containers/cup/coffeepot(src)
+		cartridge = null
+
+/obj/machinery/coffeemaker/impressa/Destroy()
+	QDEL_NULL(coffeepot)
+	QDEL_NULL(coffee)
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/Exited(atom/movable/gone, direction)
+	if(gone == coffeepot)
+		coffeepot = null
+	if(gone == coffee)
+		coffee = null
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/examine(mob/user)
+	. = ..()
+	if(coffee)
+		. += span_notice("The internal grinder conatins [coffee.len] [coffee.len == 1 ? "scoop" : "scoops"] of coffee beans")
+	return
+
+/obj/machinery/coffeemaker/impressa/update_overlays()
+	. = ..()
+	. += overlay_checks()
+
+/obj/machinery/coffeemaker/impressa/overlay_checks()
+	. = list()
+	if(coffeepot)
+		if(coffeepot.reagents.total_volume > 0)
+			. += "pot_full"
+		else
+			. += "pot_empty"
+	if(coffee_cups > 0)
+		if(coffee_cups >= max_coffee_cups/3)
+			if(coffee_cups > max_coffee_cups/1.5)
+				. += "cups_3"
+			else
+				. += "cups_2"
+		else
+			. += "cups_1"
+	if(sugar_packs)
+		. += "extras_1"
+	if(creamer_packs)
+		. += "extras_2"
+	if(sweetener_packs)
+		. += "extras_3"
+	if(coffee_amount)
+		if(coffee_amount < 0.7*BEAN_CAPACITY)
+			. += "grinder_half"
+		else
+			. += "grinder_full"
+	return
+
+/obj/machinery/coffeemaker/impressa/handle_atom_del(atom/A)
+	. = ..()
+	if(A == coffeepot)
+		coffeepot = null
+	if(A == coffee)
+		coffee.Cut()
+	update_appearance()
+
+/obj/machinery/coffeemaker/impressa/can_brew()
+	if(coffee_amount <= 0)
+		balloon_alert_to_viewers("no coffee beans added!")
+		return FALSE
+	if(!coffeepot)
+		balloon_alert_to_viewers("no coffeepot inside!")
+		return FALSE
+	if(machine_stat & (NOPOWER|BROKEN) )
+		balloon_alert_to_viewers("machine unpowered!")
+		return FALSE
+	if(coffeepot.reagents.total_volume >= coffeepot.reagents.maximum_volume)
+		balloon_alert_to_viewers("the coffeepot is already full!")
+		return FALSE
+	return TRUE
+
+/obj/machinery/coffeemaker/impressa/attackby(obj/item/attack_item, mob/living/user, params)
+	//You can only screw open empty grinder
+	if(!coffeepot && default_deconstruction_screwdriver(user, icon_state, icon_state, attack_item))
+		return
```

Comment:
Prefer to explicitly `return FALSE`.

Reply:
Addressed in [1758d26](https://github.com/tgstation/tgstation/pull/70991/commits/1758d260bf38349a463fb0c4c94b19bde1396d9f)